storage/s3.go

```diff
@@ -767,3 +917,21 @@ func (a *writeAtAdapter) Write(p []byte) (int, error) {
 	a.offset += int64(n)
 	return n, nil
 }
+
+// getRegion extracts correct region of the bucket
+// from the provided error.
+func getRegion(err error) (string, error) {
+	if awsErr, ok := err.(s3.RequestFailure); ok {
+
+		re := regexp.MustCompile("expecting '.*'")
```

Comment:
Is there any other way to get the region from an error message? I think this approach is not safe. Any error may match with `expecting  `.*'` regex (it is too generic).

Reply:
yes, indeed this regex is too generic, I will try to find a better way. thanks for pointing it out.